### PR TITLE
feat: add the 'eth_phishing_detection' logic to the connect accounts flow

### PR DESCRIPTION
**Pull request by omridan159**
## **Description**

Add the 'eth_phishing_detection' logic to the connect accounts flow

eth_phishing_detection is a utility for detecting phishing domains targeting Web3 users.
For checking why a given domain was blocked, there is a third-party search tool maintained by ChainPatrol.

Now when user connects to a dapp via SDK that's url is blacklisted in our eth_phishing_detection list, MetaMask mobile app should show a warning to the user before selecting to connect or not. This will allow MetaMask to warn the user while letting user to decide what to do next.

### WC 

https://github.com/MetaMask/metamask-mobile/assets/61094771/cacef0ac-17f6-4672-b830-450f30d2e4b4

### MM_SDK
https://github.com/MetaMask/metamask-mobile/assets/61094771/58358497-82e1-4783-891e-76d37b6a0bfd

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/core/SDKConnect/Connection/EventListenersHandlers/handleClientsReady.ts [hunk 1]
```diff
@@ -42,7 +43,10 @@ function handleClientsReady({
           });
 
           // Redirect on deeplinks
-          if (instance.trigger === 'deeplink') {
+          if (
+            instance.trigger === 'deeplink' &&
+            instance.origin !== AppConstants.DEEPLINKS.ORIGIN_QR_CODE
```
Comment (abretonc7s) [inline app/core/SDKConnect/Connection/EventListenersHandlers/handleClientsReady.ts:48]: why do we need this new condition?

**Review by abretonc7s — DISMISSED** [review]

**Review by andreahaku — DISMISSED** [review]
LGTM

**Comment by christopherferreira9** [comment]
**WalletConnect example:**

https://github.com/MetaMask/metamask-mobile/assets/104831203/b93b3989-cd56-4054-bf54-221430410236

**MetaMask SDK Example:**

https://github.com/MetaMask/metamask-mobile/assets/104831203/c256252f-9002-4902-b1a7-781e8a1e0adc

Verified that:
- All links work in the modal
- Connection to both WalletConnect and SDK are terminated once the user taps "Back to safety"
- Tested on both iOS and Android

File: app/components/Views/AccountConnect/AccountConnect.tsx [hunk 2]
```diff
@@ -442,6 +534,29 @@ const AccountConnect = (props: AccountConnectProps) => {
     ],
   );
 
+  const renderPhishingModal = () => (
```
Comment (abretonc7s) [inline app/components/Views/AccountConnect/AccountConnect.tsx:537]: add useCallback

**Review by andreahaku — DISMISSED** [review]
LGTM

**Review by abretonc7s — CHANGES_REQUESTED** [review]
small change to have consistent use of useCallback

**Review by abretonc7s — APPROVED** [review]

**Comment by christopherferreira9** [comment]
Looking good for QA.

**Pull request merged**